skoolkit/basic.py

```diff
@@ -114,14 +115,92 @@
     255: 'COPY'
 }
 
+def get_number(snapshot, i):
+    if snapshot[i] == 0:
```

Comment:
There is no need to use the 'num' variable here. Also explicitly testing whether an integer is zero is usually unpythonic.

    if snapshot[i]:
        return _get_float(snapshot, i)
    return _get_integer(snapshot, i)